source/common/http/utility.cc

```diff
@@ -319,5 +319,29 @@ void Utility::appendToHeader(HeaderString& header, const std::string& data) {
   header.append(data.c_str(), data.size());
 }
 
+void Utility::extractHostPathFromUri(const std::string& uri, std::string& host, std::string& path) {
```

Comment:
Needs a bunch of test covering expected and corner cases.

Reply:
Added test